### ISSUE TITLE
CSS Grid ( proposal) 

### ISSUE THREAD

**Issue by AndreaMonzini** [silexlabs/Silex#655]
What  about implement a visual CSS Grid Layout in Silex?

https://www.w3.org/TR/css-grid-1/#intro

Now CSS Grid is compatible with almost all browser:

https://caniuse.com/#search=css%20grid

**Comment by lexoyo** [comment]
Very interesting
What do you think the interface should look like?

**Comment by budislav** [comment]
I just wanted to post about CSS grid :) That is really awesome feature in css, we don't need bootstrap grid we have clean css grid.
I don't now is this a good for help but maybe just to give us some idea how it can look. https://www.cssgrid.cc/css-grid-builder.html
Every page can have one tab beside style and component settings for css grid and all settings for that thing there.

**Comment by AndreaMonzini** [comment]
@lexoyo I was thinking to a visual sliders on the properties panel on the right.

**Comment by lexoyo** [comment]
I just teached myself css grid and .... wow!!! 
And great video @budislav, we should use [the idea he has for the next gen wordpress editor](https://youtu.be/txZq7Laz7_4?t=33m35s) and  [he talks about website builder here](https://youtu.be/txZq7Laz7_4?t=31m10s)

Thank you guys!

> @lexoyo I was thinking to visual sliders on the properties panel on the right.

I'm not 100% sure to see what you mean... @budislav do you? (Budislav is a UI/UX designer)

We should be able to find a way to do this, easy for me and for the user :)

**Comment by budislav** [comment]
@lexoyo I would agree with him css grid is a game changer, this grid should be a base of website so we can drag and drop components on the grid fields. I think this also affect on responsive function. That gutenberg editor seems to be hard to use I don't see anything user friendly there. Webflow is much better in compare to wix for me, focus is on design, unlimited possibilities. Their purpose is same but they operate completely different. Gutenberg tries to be like wix but the good things with gutenberg are it is integrated with wordpress and it is free. 
World need one open source complete visual web designing tool with unlimited possibilities, clean html and css export. Tool with can be created web site with latest web technology. Modular so new features are never far behind. For example possibility to add blur effect on text or button with css blur.

> I'm not 100% sure to see what you mean... @budislav do you?

I think he talks about same thing as me :) One new tab on the right side with css grid settings. This is great example https://alialaa.github.io/css-grid-cheat-sheet/ thanks @AndreaMonzini. Also nice tutorials https://mozilladevelopers.github.io/playground/css-grid and https://css-tricks.com/snippets/css/complete-guide-grid/ unfortunately I don't have much time to explore in depth..

**Comment by AndreaMonzini** [comment]
@lexoyo @budislav yes, i was thinking about CSS Grid settings on the right Panel and i like sliders to select numeric values but they are not essentials.
About the resources, thanks to @valentinogagliardi who collected the links and thanks to the authors of the CSS Grid projects and resources.

With CSS Grid should be possible to manage responsive website and therefore mobile version of the website, right?

**Comment by budislav** [comment]
@valentinogagliardi thanks for the resources they are really helpful! I really like tools for visual playing with css grid.

> With CSS Grid should be possible to manage responsive website and therefore mobile version of the website, right?

I believe yes. What do you think @lexoyo?

**Comment by lexoyo** [comment]
As I understand it, for each element in a 'grided' container, we would be able to set a position in the grid for desktop and another position for mobile... but I'm not sure how convenient it would be to have to set them both

**Comment by budislav** [comment]
A sneak peek at Grid in Webflow
https://www.youtube.com/watch?v=jpJE25aq7tA&t=2421s

**Comment by lexoyo** [comment]
Wow that's a great demo
